modules/ingester/instance.go

```diff
@@ -402,17 +402,21 @@ func (i *instance) ClearFlushedBlocks(completeBlockTimeout time.Duration) error
 	return err
 }
 
-func (i *instance) FindTraceByID(ctx context.Context, id []byte, allowPartialTrace bool) (*tempopb.Trace, error) {
+func (i *instance) FindTraceByID(ctx context.Context, id []byte, allowPartialTrace bool) (*tempopb.TraceByIDResponse, error) {
 	ctx, span := tracer.Start(ctx, "instance.FindTraceByID")
 	defer span.End()
 
 	var err error
 	var completeTrace *tempopb.Trace
+	metrics := tempopb.TraceByIDMetrics{}
 
 	// live traces
 	i.tracesMtx.Lock()
 	if liveTrace, ok := i.traces[i.tokenForTraceID(id)]; ok {
 		completeTrace, err = model.MustNewSegmentDecoder(model.CurrentEncoding).PrepareForRead(liveTrace.batches)
+		for _, b := range liveTrace.batches {
```

Comment:
this is all in memory. i don't think we want to count this is "bytes inspected". i think we also have the question of whether to count any of this disk access in the final calculation.

let's sync out of band on this.

Reply:
after some discussion we decided to include these bytes read as well. we can revisit later if needed